server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -0,0 +1,129 @@
+/* 
+ * Description: File for building ads and adding them to Firestore
+ * Author: Kira Toal
+ * Date: June 24, 2020
+ */ 
+package com.google.sps.utils;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.sps.utils.Ad.Builder;
+import com.google.sps.utils.Ad;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AdRowProcessor {
+
+  private static final String PATH_TO_SERVICE_ACCOUNT = "./serviceAccountKey.json"; 
+  private static final String DATABASE_URL = "https://step9-2020-capstone.firebaseio.com"; 
+  private String[] row; 
+
+  public AdRowProcessor(String[] csvRow) throws Exception {
+    this.row = csvRow; 
+    // initialize app
+    FileInputStream serviceAccount = new FileInputStream(PATH_TO_SERVICE_ACCOUNT);
+    FirebaseOptions options = new FirebaseOptions.Builder()
+        .setCredentials(GoogleCredentials.fromStream(serviceAccount))
+        .setDatabaseUrl(DATABASE_URL)
+        .build();
+    if(FirebaseApp.getApps().isEmpty()) {
+      FirebaseApp.initializeApp(options);
+    }
+  }
+
+  public void addAdToDatabase(Ad ad, int rowIndex, String collection) throws Exception {
```

Comment:
writeAd is easier to say.  Also more accurate since this isn't just add, right, it's also update.

Reply:
Updated name